src/modules/commander/Commander.cpp

```diff
@@ -2731,22 +2731,25 @@ Commander::set_main_state_rc(const vehicle_status_s &status_local, bool *changed
 	// feature, just in case offboard control goes crazy.
 
 	const bool position_got_valid = !_last_condition_global_position_valid && status_flags.condition_global_position_valid;
-	const bool had_rc_before = _last_sp_man.timestamp != 0;
-	const bool rc_values_not_updated = _last_sp_man.timestamp == sp_man.timestamp;
-	const bool all_switches_stayed =
-		(_last_sp_man.offboard_switch == sp_man.offboard_switch)
-		&& (_last_sp_man.return_switch == sp_man.return_switch)
-		&& (_last_sp_man.mode_switch == sp_man.mode_switch)
-		&& (_last_sp_man.acro_switch == sp_man.acro_switch)
-		&& (_last_sp_man.rattitude_switch == sp_man.rattitude_switch)
-		&& (_last_sp_man.posctl_switch == sp_man.posctl_switch)
-		&& (_last_sp_man.loiter_switch == sp_man.loiter_switch)
-		&& (_last_sp_man.mode_slot == sp_man.mode_slot)
-		&& (_last_sp_man.stab_switch == sp_man.stab_switch)
-		&& (_last_sp_man.man_switch == sp_man.man_switch);
-
-	/* manual setpoint has not updated, do not re-evaluate it */
-	if (!position_got_valid && ((had_rc_before && rc_values_not_updated) || all_switches_stayed)) {
+	const bool first_time_rc = _last_sp_man.timestamp == 0;
```

Comment:
This works because the `set_main_state_rc` call is conditioned by `sp_man.timestamp != 0`.
For consistency I'd consider adding that check here as well.

Reply:
Thanks for checking!
You're right. There's even a duplicate such condition here: https://github.com/PX4/Firmware/blob/903deb7579ec655ffea526eb5701b78f9f5c9ce8/src/modules/commander/Commander.cpp#L2084 I'll quickly add this.

Sadly there are more potential corner cases that start with "It works because it's conditioned..." E.g. `_last_condition_global_position_valid` only gets updated when you have RC and `_last_sp_man` only gets updated when you evaluate the mode switch or are in a manual mode. I want to refactor to get rid of these obfuscated cases but not before the release. I hope it's ok to leave it for this pr.

Reply:
I always like brackets around conditions, I think it's more readable:

```suggestion
	const bool first_time_rc = (_last_sp_man.timestamp == 0);
```

Reply:
I'm also a fan of brackets if there are more than one operator and two operands on the right side of the assignment.